symengine/visitor.h

```diff
@@ -72,10 +72,14 @@ void postorder_traversal_stop(const Basic &b, StopVisitor &v);
 class HasSymbolVisitor : public BaseVisitor<HasSymbolVisitor, StopVisitor>
 {
 protected:
-    RCP<const Symbol> x_;
+    Ptr<const Symbol> x_;
     bool has_;
 
 public:
+    HasSymbolVisitor(const Symbol &x) : x_(&x)
```

Comment:
If you want to make this even cleaner, you can pass in `Ptr<const Symbol> x`, then you just assign Ptr instances, which is safe and clean. And you move up the responsibility of making sure the Symbol doesn't get deallocated one level up --- into `has_symbol`, which is the place which knows that Symbol won't go out of scope.